miniapps/meshing/pmesh-optimizer.cpp

```diff
@@ -323,7 +333,8 @@ int main (int argc, char *argv[])
       mesh_name << "perturbed.mesh";
       ofstream mesh_ofs(mesh_name.str().c_str());
       mesh_ofs.precision(8);
-      pmesh->PrintAsOne(mesh_ofs);
+      //x.HostRead();
+      //pmesh->PrintAsOne(mesh_ofs);
```

Comment:
Is there an issue with PrintAsOne( )?

Reply:
Yes, the data has to be brought back on the host.

Reply:
I think I fixed that in a recent PR.

Reply:
Actually, https://github.com/mfem/mfem/pull/1576 fixes `ParGridFunction::SaveAsOne`. The fix for `PrintAsOne` should be added in this PR, if the fix for `SaveAsOne` is not sufficient.